packages/plugin-npm/sources/npmPublishUtils.ts

```diff
@@ -45,7 +48,7 @@ export async function makePublishBody(workspace: Workspace, buffer: Buffer, {acc
         length: buffer.length,
       },
     },
-
+    gitHead: await gitHead(workspace.cwd),
```

Comment:
It'd be better to have this as an argument to make publish body rather than invoked from within the function. Publish utils is part of the public API and my [project](https://tophat.github.io/monodeploy/) uses it. I'd rather have the option to specify a SHA rather than have it assume. I will note it does seem an odd coupling with git.

Reply:
That's a good suggestion. Thank you. I have done that.
I have also updated the PR to use git cli for getting the commitId.